rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -136,9 +137,24 @@ NodeGraph::get_node_names() const
   std::vector<std::string> nodes;
   auto names_and_namespaces = get_node_names_and_namespaces();
 
-  for (const auto & it : names_and_namespaces) {
-    nodes.push_back(it.first);
-  }
+  std::transform(names_and_namespaces.begin(),
+    names_and_namespaces.end(),
+    std::back_inserter(nodes),
+    [](std::pair<std::string, std::string> nns) {
+      std::string return_string;
+      if ((nns.second.back() == '/') || (nns.first.front() == '/')) {
```

Comment:
If both of these are true wouldn't you get `/ns//name`?

Reply:
You are correct. I will modify the return_string to search and replace any "//" substrings. That should probably help.

I added these checks in because the unit tests were entering "/" as a namespace, and I wanted to make sure the output was at least qualified. Ideally, the name and namespace would both be enforced to be correct on node creation, but I figured that's beyond the scope of this change.

Reply:
They should be enforced, and I think the name may never have a leading slash (I'd have to check), but the name space may have a slash in the `back()` position, specifically for the case where the namespace is `/`.

Reply:
That makes my life even easier, assuming the name never has a leading slash. Also makes the code more performant (eww, string operations, eww)